src/components/Actions.js

```diff
@@ -0,0 +1,95 @@
+import { useCallback, useEffect, useState } from 'react';
+import { Navigation, Popup } from "./index.js";
+
+export function Actions ({data, setData, currentIndex, setCurrentIndex, setCommentTitle, style}) {
+    const [popoverOpen, setPopoverOpen] = useState(false);
+
+    const handleAccept = useCallback(() => {
+        let comment = data[currentIndex].comment;
+        if ( data[currentIndex].status === 'FAILED' && comment.length > 0 ) {
+            setCommentTitle('Comment from Previous Failure');
+        }
+
+        setData(prevNewData => prevNewData.map((data, i) => {
+          if (i === currentIndex) {
+            return {
+              ...data,
+              status: 'PASSED',
+              comment: comment
+            }
+          }
+          return data;
+        }));
+      }, [currentIndex, data, setData, setCommentTitle]);
+    
+    const handleReject = useCallback((comment) => {
+        setData(prevNewData => prevNewData.map((data, i) => {
```

Comment:
**Suggestion (non-blocking)**: Should we add a line here to reset the comment title back to the failure version? If you reject, then accept, then re-reject, the header "Comment from Preview Failure" still remains